src/v/cluster/topics_frontend.cc

```diff
@@ -1007,7 +1007,9 @@ ss::future<topic_result> topics_frontend::do_delete_topic(
 }
 
 ss::future<topic_result> topics_frontend::purged_topic(
-  nt_revision topic, model::timeout_clock::duration timeout) {
+  nt_revision topic,
+  topic_purge_domain domain,
+  model::timeout_clock::duration timeout) {
```

Comment:
> Am I understanding correctly that this PR isn't going to purge data yet?

It is, data is purged by the catalog. (Side note, this is probably not reliable enough so eventually we'll have to implement our own purger).

> I.e. would you mind deferring this commit until we actually purge data? For instance, in my mind, purging Iceberg data could be entirely decoupled from the archival subsystem (as is the rest of Iceberg/datalake).

We can't defer this commit because we need some record in topic table for deleted topics saying "we still need to delete the iceberg table"

Reply:
>We can't defer this commit because we need some record in topic table for deleted topics saying "we still need to delete the iceberg table"

Can you elaborate on why it needs to be in the topic table? In my mind, information about the Iceberg tables ought to be in the coordinator and/or its stm. Why isn't that sufficient?

Reply:
Consider the following scenario:
1. redpanda.iceberg.delete is initially false
2. user sets redpanda.iceberg.delete to true
3. immediately afterwards user deletes the table

We have to ensure that the "we need to delete iceberg table for this topic" bit of information is not lost - iceberg tombstone is exactly this. Since updates in 2 and 3 can be very fast, and nothing really blocks them, we might not have time to transfer this info to the coordinator state while the topic is still alive.

Reply:
Ah I see. I think this problem goes away if we make table creation a function of the coordinator, is that right?

For instance if the coordinator handled creation, if the coordinator never created the table, there wouldn't be a table to delete.

If that's the main limitation, I'd much rather bake table lifecycle into the coordinator than the controller.